modules/templates/vars/vars.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package vars
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ErrWrongSyntax represents a wrong syntax with a tempate
+type ErrWrongSyntax struct {
+	Template string
+}
+
+func (err ErrWrongSyntax) Error() string {
+	return fmt.Sprintf("Wrong syntax found in %s", err.Template)
+}
+
+// IsErrWrongSyntax returns true if the error is ErrWrongSyntax
+func IsErrWrongSyntax(err error) bool {
+	_, ok := err.(ErrWrongSyntax)
+	return ok
+}
+
+// ErrNoMatchedVar represents an error that no matched vars
+type ErrNoMatchedVar struct {
+	Template string
+	Var      string
+}
+
+func (err ErrNoMatchedVar) Error() string {
+	return fmt.Sprintf("No matched variable %s found for %s", err.Var, err.Template)
+}
+
+// IsErrNoMatchedVar returns true if the error is ErrNoMatchedVar
+func IsErrNoMatchedVar(err error) bool {
+	_, ok := err.(ErrNoMatchedVar)
+	return ok
+}
+
+// Expand replaces all variables like {var} to match
+func Expand(template string, match map[string]string, subs ...string) (string, error) {
```

Comment:
You could save all the extra trouble with the positional arguments by using something like
```suggestion
// All positional arguments will be converted into corresponding map entries: match["x"] = subs[x])
func Expand(template string, match map[string]string, subs ...string) (string, error) {
  for index, substitution := range subs {
    match[strconv.Itoa(index)] = substitution
  }
```